src/main/webapp/2Dscript.js

```diff
@@ -80,7 +151,279 @@ function addCenterControl(controlDiv) {
+    keyEntry.innerText = key;
+    currentRow.appendChild(keyEntry);
+
+    let checkBoxEntry = document.createElement("td");
+    const input = document.createElement("input");
+    input.type = "checkbox";
+    input.id = key;
+    input.addEventListener("click", function (event) {
+      // Get the necessary content for this runId if not stored in the local cache.
+      console.log("Checkbox event");
+      const checkboxElement = event.target;
+      console.log("The id is:" + event.target.id);
+      console.log("The event is" + event.target.toString());
+      console.log("The checkbox is " + this.checked);
+      if (this.checked) {
+        // First check the cache for this value.
+        dataEntries = datas[event.target.id];
+        console.log("The value of dataentries is " + dataEntries);
+        // if the value is not found in the cache then fetch it.
+        if (dataEntries === undefined) {
+          console.log("Data Entries was false");
+          console.log("The id is:" + event.target.id);
+          fetch("/getrun?id=" + event.target.id + "&dataType=pose").then(response => response.json())
+            .then(data => dataEntries = data).then(() => {
+              datas[event.target.id] = dataEntries;
+              const toGraph = plotLine(dataEntries);
+              maps[event.target.id] = toGraph;
+              toGraph.setMap(map);
+            });
+        }
+        else {
+          // Still plot the line, doesn't need to be asynchronous.
+          // TODO(morleyd): abstract this out into a function.
+          console.log("Dataentries was true");
+          const toGraph = plotLine(dataEntries);
+          maps[event.target.id] = toGraph
+          toGraph.setMap(map);
+        }
+      }
+      else {
+        // In this case the box has become unchecked. We want to remove this graph.
+        // We can only remove this box if it has been generated before and then can be removed appropriately.
+        const toRemove = maps[event.target.id];
+        if (toRemove)
+          toRemove.setMap(null);
+      }
+
+    });
+    checkBoxEntry.appendChild(input);
+    currentRow.appendChild(checkBoxEntry);
+    table.appendChild(currentRow);
+    console.log(currentRow);
+    even = !even;
+
+  }
+  return table;
+
+}
+
+function plotLine(dataEntries) {
+  let currentLine = [];
+  // now add the map with the selected color of the element
+  for (let i = 0; i < dataEntries.length; i++) {
+    currentLine.push({ lat: dataEntries[i].lat, lng: dataEntries[i].lng });
+  }
+  console.log(currentLine);
+  currentLineGraph = new google.maps.Polyline({
+    path: currentLine,
+    geodesic: true,
+    strokeColor: 'blue',
+    strokeOpacity: 1.0,
+    strokeWeight: 2
+  });
+  flightPath.setMap(null);
+  console.log("The value of map is " + map);
+  return currentLineGraph;
+}
+
+// When finished drawing the box, instead of leaving the widget, create markers at the current lat and intersection with the line.
+$(function () {
```

Comment:
This is another huge and deeply nested function, please refactor. I am not familiar with the `$` syntax, what is it doing?

Reply:
I used Jquery in this code to simplify the many nested event handlers $( gives a global selector in jquery